docs/topics/compatibility-guides/compatibility-guide-22.md

```diff
@@ -0,0 +1,341 @@
+
+## Standard library
+
+### Deprecate kotlin.native.Throws
+
+> **Issue**: [KT-72137](https://youtrack.jetbrains.com/issue/KT-72137)
+>
+> **Component**: Kotlin/Native
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: `kotlin.native.Throws` is deprecated; use the common `kotlin.Throws` annotation instead. 
+>
+> **Deprecation cycle**:
+>
+> - 1.9.0: report a warning when using `kotlin.native.Throws`
+> - 2.2.0: raise the warning to an error
+
+### Deprecate AbstractDoubleTimeSource
+
+> **Issue**: [KT-72137](https://youtrack.jetbrains.com/issue/KT-72137)
+>
+> **Component**: kotlin-stdlib
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: `AbstractDoubleTimeSource` is deprecated; use `AbstractLongTimeSource` instead.
+>
+> **Deprecation cycle**:
+>
+> - 1.8.20: report a warning when using `AbstractDoubleTimeSource`
+> - 2.2.0: raise the warning to an error
+
+## Tools
+
+### Correct the setSource() function in KotlinCompileTool to replace sources instead of adding to them
+
+> **Issue**: [KT-59632](https://youtrack.jetbrains.com/issue/KT-59632)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: behavioral
+>
+> **Short summary**: The [`setSource()`](https://kotlinlang.org/api/kotlin-gradle-plugin/kotlin-gradle-plugin-api/org.jetbrains.kotlin.gradle.tasks/-kotlin-compile-tool/set-source.html#) function in [`KotlinCompileTool`](https://kotlinlang.org/api/kotlin-gradle-plugin/kotlin-gradle-plugin-api/org.jetbrains.kotlin.gradle.tasks/-kotlin-compile-tool/#) now replaces configured sources instead of adding to them.
+> If you want to add sources without replacing existing ones, use the [`source()`](https://kotlinlang.org/api/kotlin-gradle-plugin/kotlin-gradle-plugin-api/org.jetbrains.kotlin.gradle.tasks/-kotlin-compile-tool/source.html#) function.
+>
+> **Deprecation cycle**:
+>
+> - 2.2.0: enable new behavior
+
+### Deprecate KotlinCompilationOutput#resourcesDirProvider
+
+> **Issue**: [KT-70620](https://youtrack.jetbrains.com/issue/KT-70620)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The `KotlinCompilationOutput#resourcesDirProvider` field is deprecated.
+> Use `KotlinSourceSet.resources` in your Gradle build script instead to add additional resource directories.
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: `KotlinCompilationOutput#resourcesDirProvider` is deprecated with a warning
+> - 2.2.0: raise the warning to an error
+
+### Change the type of annotationProcessorOptionProviders to use CommandLineArgumentProvider
+
+> **Issue**: [KT-58009](https://youtrack.jetbrains.com/issue/KT-58009)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The type of [`annotationProcessorOptionProviders`](https://kotlinlang.org/api/kotlin-gradle-plugin/kotlin-gradle-plugin-api/org.jetbrains.kotlin.gradle.tasks/-base-kapt/annotation-processor-option-providers.html#) in `BaseKapt` is changed from `MutableList<Any>`
```

Comment:
Not true - `BaseKapt.annotationProcessorOptionProviders` still has `MutableList<Any>` type and this property is deprecated.

Users should migrate to `BaseKapt.annotationProcessorOptionsProviders` (note: option**s**) which has `ListProperty<CommandLineArgumentProvider>` type.